101-data-factory-blob-to-sql-copy/azuredeploy.json

```diff
@@ -0,0 +1,222 @@
+      "storageAccountName": {
+        "type": "string",
+        "metadata": {
+          "description":  "Azure Storage account name"
+        }
+      },
+      "storageAccountKey": {
+        "type": "securestring",
+        "metadata": {
+          "description":  "Azure Storage account key"
+        }
+      },
+      "sourceBlobContainer": {
+        "type": "string",
+        "metadata": {
+          "description":  "Azure blob container name"
+        }
+      },
+      "sourceBlobName": {
+        "type": "string",
+        "metadata": {
+          "description":  "Azure blob name"
+        }
+      },
+      "sqlServerName": {
+        "type": "string",
+        "metadata": {
+          "description":  "Azure SQL server name"
+        }
+      },
+      "databaseName": {
+        "type": "string",
+        "metadata": {
+          "description":  "Azure SQL database name"
+        }
+      },
+      "sqlServerUserName": {
+        "type": "string",
+        "metadata": {
+          "description":  "Azure SQL server - user name"
+        }
+      },
+      "sqlServerPassword": {
+        "type": "securestring",
+        "metadata": {
+          "description":  "Azure SQL server - user password"
+        }
+      },
+      "targetSQLTable": {
+        "type": "string",
+        "metadata": {
+          "description":  "Azure SQL table name"
+        } 
+      }
+    },
+    "variables": {
+      "apiVersion": "2015-10-01",
+      "storageApiVersion" :  "2015-05-01-preview",
+      "azureSqlLinkedServiceName": "AzureSqlLinkedService",
+      "azureStorageLinkedServiceName": "AzureStorageLinkedService",
+      "blobInputDatasetName": "BlobInputDataset",
+      "sqlOutputDatasetName": "SQLOutputDataset",
+      "pipelineName": "Blob2SQLPipeline"    
+    },
+    "resources": [
+      {
+        "name": "[parameters('dataFactoryName')]",
+        "apiVersion": "[variables('apiVersion')]",
+        "type": "Microsoft.DataFactory/datafactories",
+        "location": "West US",
+        "resources": [
+          {
+            "type": "Microsoft.DataFactory/datafactories/linkedservices",
+            "name": "[concat(parameters('dataFactoryName'), '/', variables('azureStorageLinkedServiceName'))]",
+            "dependsOn": [ "[concat('Microsoft.DataFactory/dataFactories/', parameters('dataFactoryName'))]" ],
+            "apiVersion": "[variables('apiVersion')]",
+            "properties": {
+              "type": "AzureStorage",
+              "description": "Azure Storage linked service",
+              "typeProperties": {
+                "connectionString": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('storageAccountName'),';AccountKey=',parameters('storageAccountKey'))]"
+              }
+            }
+          },
+          {
+            "type": "Microsoft.DataFactory/datafactories/linkedservices",
+            "name": "[concat(parameters('dataFactoryName'), '/', variables('azureSqlLinkedServiceName'))]",
```

Comment:
since these are child resources, remove the extra name segments to improve readability.